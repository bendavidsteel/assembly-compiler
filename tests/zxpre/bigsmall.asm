small:    db "too small\n"
big:      db "big enough\n"

          ; Compute 1 + 2 + 3 + 4 + 5
main:     mov %a,0x00             ; total = 0
          mov %b,0x01             ; i = 1
lbegin:   add %a,%b              ; total += i
          add %b,0x01             ; i++
          cmp %b,0x05             ; if(b < 5)
          jl  lbegin             ;   goto lbegin

          ; Print appropriate mesage
          mov %a,0x04             ; sys_write
          mov %b,0x01             ; stdout
          cmp %a,0x0b             ; if(total <= 11)
          jle small              ;    goto psmall
          mov %c,big             ; big string
          mov %d,0x0b             ; length 11
          jmp print              ; goto print
psmall:   mov %c,small           ; small string
          mov %d,0x0a             ; length 10
print:    syscall                ; actually print!

          ; Clean exit
          mov %a,0x01             ; sys_exit
          syscall                ; invoke sys call
