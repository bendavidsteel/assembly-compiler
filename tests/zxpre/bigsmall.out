small:   db "too small\n"
big:     db "big enough\n"
main:    mov %a,0x00
         mov %b,0x01
lbegin:  add %a,%b
         add %b,0x01
         cmp %b,0x05
         jl lbegin
         mov %a,0x04
         mov %b,0x01
         cmp %a,0x0b
         jle small
         mov %c,big
         mov %d,0x0b
         jmp print
psmall:  mov %c,small
         mov %d,0x0a
print:   syscall
         mov %a,0x01
         syscall
